lib/src/firestore_cache.dart

```diff
@@ -43,12 +44,13 @@ class FirestoreCache {
   /// This method should only be used if the document you are fetching does not
   /// change over time. Once the document is cached, it will always be read from
   /// the cache.
-  static Future<DocumentSnapshot<Map<String, dynamic>>> getDocument(
-    DocumentReference<Map<String, dynamic>> docRef, {
+  static Future<DocumentSnapshot<T>> getDocument<T>(
+    DocumentReference<T> docRef, {
```

Comment:
What are the scenarios where someone can pass in a `DocumentReference<T>`? I see that from `Firestore.doc` [method](https://pub.dev/documentation/cloud_firestore/latest/cloud_firestore/FirebaseFirestore/doc.html) only returns `DocumentReference<Map<String, dynamic>>>`, unless I’m missing something?

Reply:
Sorry I missed the snippet you shared, let me read through the code again.